crates/wasmparser/src/readers/core/operators.rs

```diff
@@ -1176,3 +1187,1280 @@ impl<'a> Iterator for OperatorsIteratorWithOffsets<'a> {
         Some(result)
     }
 }
+
+/// Trait implemented by types that can visit all [`Operator`] variants.
+#[allow(missing_docs)]
+pub trait VisitOperator<'a, Input> {
+    /// The result type of the visitor.
+    type Output: 'a;
+
+    /// Visits the [`Operator`] `op` using the given `input`.
+    ///
+    /// # Note
+    ///
+    /// This is a convenience method that is intended for non-performance
+    /// critical use cases. For performance critical implementations users
+    /// are recommended to directly use the respective `visit` methods or
+    /// implement [`VisitOperator`] on their own.
+    #[inline(always)]
```

Comment:
This is such a big method I think it's best to leave this to LLVM, not force it here.

Reply:
As weirdly as it sounds but I was able to resolve a performance regression using this `#[inline(always)]` annotation. Remove it?